Source/WebCore/page/LocalDOMWindow.cpp

```diff
@@ -640,9 +640,14 @@ ExceptionOr<Ref<NodeList>> LocalDOMWindow::collectMatchingElementsInFlatTree(Nod
     SelectorQuery& query = queryOrException.releaseReturnValue();
 
     Vector<Ref<Element>> result;
-    for (auto& node : composedTreeDescendants(*scopeContainer)) {
-        if (RefPtr element = dynamicDowncast<Element>(node); element && query.matches(*element) && !node.isInUserAgentShadowTree())
+    auto descendants = composedTreeDescendants(*scopeContainer);
+    auto it = descendants.begin();
```

Comment:
Unclear why the new code looks so much worse here.

Reply:
I had some difficulty reffing  the node and at the same time appending correctly to result, hence the complicated code. But I think we can keep it simpler and still fix the warning, by simply creating a StaticNodeList instead of a StaticElementList, since in the end we return a NodeList.

Reply:
I doubt these changes are needed, can you please explain what compiler error you were getting?